core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1651,6 +1659,9 @@ struct TestIsFinite {
 };
 
 TEST(TEST_CATEGORY, mathematical_functions_isfinite) {
+#if __FINITE_MATH_ONLY__
+  GTEST_SKIP() << "skipping when compiling with -ffinite-math-only";
+#endif
```

Comment:
Why not run the `infinite(FINITE)`?

Reply:
Originally, I tried dealing with all warnings (and not use `-Wno-nan-infinity-disabled`) by disabling/guarding everything that would trigger the warning. Since every invocation of `isinf` triggered it, I disabled the whole test.
We could try to be more selective.